sdk/dotnet/Pulumi/Deployment/Deployment_Invoke.cs

```diff
@@ -16,21 +18,70 @@ Task IDeployment.InvokeAsync(string token, InvokeArgs args, InvokeOptions? optio
         Task<T> IDeployment.InvokeAsync<T>(string token, InvokeArgs args, InvokeOptions? options)
             => InvokeAsync<T>(token, args, options, convertResult: true);
 
+        Output<T> IDeployment.Invoke<T>(string token, InvokeArgs args, InvokeOptions? options)
+            => new Output<T>(RawInvoke<T>(token, args, options));
+
+        private async Task<OutputData<T>> RawInvoke<T>(string token, InvokeArgs args, InvokeOptions? options)
+        {
+            // This method backs all `Fn.Invoke()` calls that generate
+            // `Output<T>` and may include `Input<T>` values in the
+            // `args`. It needs to decide which control-flow tracking
+            // features are supported in the SDK and which ones in the
+            // provider implementing the invoke logic.
+            //
+            // Current choices are:
+            //
+            // - any resource dependency found by a recursive
+            //   traversal of `args` that awaits and inspects every
+            //   `Input<T>` will always be propagated into the
+            //   `Output<T>`; the provider cannot "swallow"
+            //   dependencies
+            //
+            // - the provider is responsible for deciding whether the
+            //   `Output<T>` is secret and known, and may add
+            //   additional dependencies
+            //
+            // This means that presence of secrets or unknowns in the
+            // `args` does not guarantee the result is secret or
+            // unknown, which differs from Pulumi SDKs that choose to
+            // implement these invokes via `Apply` (currently Go and
+            // Python).
```

Comment:
I thought Node.js will use `.apply`? Or is this comment just because the Node.js functionality hasn't landed yet?

This difference across language runtimes is unfortunate. Remind me again why we can't implement this via `Apply` in .NET?

Reply:
Answered in another comment below. LMK if you think you see a way forward with Apply. If we could deeply cross-convert plain and inputty types we still use Apply.  It's looking like quite bit of work though. I considered doing it by serializing and deserializing but at first glance even that is non-trivial with current API.